kasa/deviceconfig.py

```diff
@@ -117,9 +117,7 @@ class DeviceConfig:
     host: str
     timeout: Optional[int] = DEFAULT_TIMEOUT
     port_override: Optional[int] = None
-    credentials: Credentials = field(
-        default_factory=lambda: Credentials(username="", password="")
-    )
+    credentials: Credentials = field(default_factory=lambda: Credentials())
```

Comment:
I like your suggestion [here](https://github.com/home-assistant/core/pull/105143#discussion_r1439889873) to give back the auth hashes and store those so I plan to make credentials here optional and add an optional `auth_hash` parameter.  Could you make `credentials` optional in this PR so we're in sync?

Reply:
Maybe we could merge this as-is after removing the dunder magic for `__bool__`, and you can adjust the credentials_hash PR as you seem fit?

Reply:
Ok sure. If you can merge this tonight I’ll rebase and address your comments on the credentials_hash first thing tomorrow am.